File: src/web apps/Web/src/app/modulos/modulos-principais/veiculo/models/veiculo.ts
export class Veiculo
{
    public id: string;
    public condutorId: string;
    public placa: string;
    public modelo: string;
    public marca: string;
    public cor: string;
    public anoFabricacao: number;
    public cpf: string;
    public condutores: CondutorVeiculo[] = []
}    

class CondutorVeiculo
{
    public veiculoId: string;
    public condutorId: string;
    public cpf: string;
}

File: src/services/TP.Veiculos.Domain/Veiculo.cs
using System.Collections.Generic;
using TP.Core.DomainObjects;

namespace TP.Veiculos.Domain
{
    public class Veiculo : Entity, IAggregateRoot
    {
        public string Placa { get; private set; }
        public string Modelo { get; private set; }
        public string Marca { get; private set; }
        public string Cor { get; private set; }
        public int AnoFabricacao { get; private set; }

        private readonly List<Condutor> _condutor;
        public IReadOnlyCollection<Condutor> Condutor => _condutor;

        public Veiculo(string placa, string modelo, string marca, string cor, int anoFabricacao)
        {
            Placa = placa;
            Modelo = modelo;
            Marca = marca;
            Cor = cor;
            AnoFabricacao = anoFabricacao;
            _condutor ??= new List<Condutor>();
        }

        // EF Rel.
        protected Veiculo() { }

        public Veiculo AdicionarCondutor(Veiculo veiculo, string idCondutor, string cpf)
        {
            var condutor = new Condutor(veiculo.Id, idCondutor, cpf);

            veiculo._condutor.Add(condutor);

            return veiculo;
        }
    }
}
